// __BEGIN_LICENSE__
// Copyright (C) 2006-2010 United States Government as represented by
// the Administrator of the National Aeronautics and Space Administration.
// All Rights Reserved.
// __END_LICENSE__

package vw.platefile;

// The TileRecord store metadata about the tile itself as well as the
// information needed to place the tile back into the index tree.  The
// [col, row, level, transaction_id] info can be used to reconstruct the index
// by reading TileRecords in the blob files.
message TileHeader {
  required uint32 col = 5;
  required uint32 row = 6;
  required uint32 level = 7;
  optional uint32 transaction_id = 8 [ default = 0 ];
  optional string filetype = 10 [ default = 'jpg' ];
}

message IndexHeader {
  optional uint32 tile_size = 2;     // tile width/heigh in pixels
  optional string tile_filetype = 3; // file type [e.g. tif, png, or jpg]
  optional int32 pixel_format = 4;   // see vw/Image/PixelTypeInfo.h
  optional int32 channel_type = 5;   // see vw/Image/PixelTypeInfo.h
  optional string type = 6;          // platefile type [ toast, kml, or gigapan ]
  optional string description = 7;   // textual description

  // These are private and should be considered read-only
  optional uint32 platefile_id = 1;  // a unique number that identifies this platefile
  optional uint32 num_levels = 8;     // number of pyramid levels
  optional uint32 version = 10 [default = 1];
  optional uint32 transaction_read_cursor = 11 [default = 0];
  optional uint32 transaction_write_cursor = 12 [default = 0];
}
